c10/cuda/CUDACachingAllocator.cpp

```diff
@@ -245,12 +249,14 @@ struct AllocParams {
   AllocParams(
       int device,
       size_t size,
+      size_t requested_size,
```

Comment:
Might be simpler to just initialized requested_size when the block is set to allocated rather than when a block is created.